capstone/src/main/webapp/scripts/post.js

```diff
@@ -1,27 +1,75 @@
 function init() {
-  fetchBlobstoreUrlAndShowForm();
   loadPosts();
   getPollOptions();
-
+  fetchBlobstoreUrlAndShowForm();
 }
 
+let postResponse;
 function loadPosts() {
-  fetch('/group-post').then(response => response.json()).then((posts) => {
+  fetch('/group-post').then(response => response.json()).then((postsResp) => {
+    postResponse = postsResp;
+    const posts = postResponse["posts"];
     const allPostsList = document.getElementById('posts-container');
     allPostsList.innerHTML = '';
-    for (var i = 0; i < posts.length; i++) {
-      allPostsList.appendChild(createSinglePost(posts[i]));
+    for (let i = 0; i < posts.length; i++) {
+      allPostsList.appendChild(createSinglePost(posts[i], postResponse["likedPosts"]));
+    }
+  }).then(() => {
+    addCommentInputListener();
+    addLikeButtonListener();
+  });
+}
+
+function addCommentInputListener() {
+  let elements = document.getElementsByClassName('post-btn align-vertical comment-btn');
+    for (let i = 0; i < elements.length; i++) {
+      elements[i].addEventListener("click", function() {
+        postComment(this.id, this.id + "comment-input")
+      });
     }
+}
+
+function addLikeButtonListener() {
+  let likeBtns = document.getElementsByClassName('like-icon vertical-align');
+    for (let i = 0; i < likeBtns.length; i++) {
+      likeBtns[i].addEventListener("click", function() {
+        let postId = parseInt(this.id.substring(0, this.id.length - 4));
+        let userLikedPosts = postResponse["likedPosts"];
+        if(userLikedPosts.includes(postId)) {
+          likeToggled(this.id, false);
+        } else {
+          likeToggled(this.id, true);
+        }
+      });
+    }
+}
+
+function likeToggled(likeId, liked) {
+  let postId = likeId.substring(0, likeId.length - 4);
```

Comment:
I'd still like to see something to avoid this magic number 4, or explain it somehow in the code as a preference with a well named const or some other in-code explanation, or in a comment if there is no clever way to cover that.

Reply:
Not sure if you saw this but I asked the same earlier and she commented above that 
> I'm constructing the likeId to be the postId + "like" so I'm using substring to just get the postId.

So yeah I also think using a const in place of 4 might help clear it up?

Reply:
added a comment to explain the use of substring/4